backend/src/v2/component/launcher_v2.go

```diff
@@ -562,51 +562,53 @@ func downloadArtifacts(ctx context.Context, executorInput *pipelinespec.Executor
 		if len(artifactList.Artifacts) == 0 {
 			continue
 		}
-		inputArtifact := artifactList.Artifacts[0]
+		for _, artifact := range artifactList.Artifacts {
```

Comment:
Working on the same thing in a different [place](https://github.com/kubeflow/pipelines/pull/11762/files#diff-8ad39c78b8d631f0ca74f79f5090a722aeef939a511aa19b116e361c1a2a01c2R548), heh.